### PR TITLE
Test for "delete cluster"

### PR DESCRIPTION
**Pull request by marians**
This adds proper unit tests for the "delete cluster" command. As with "create cluster" before, business logic and result printing are now separate.

### CODE REVIEW

File: commands/delete_cluster.go [hunk 1]
```diff
@@ -43,58 +69,104 @@ func init() {
 	DeleteCommand.AddCommand(DeleteClusterCommand)
 }
 
-// checks preconditions
-func checkDeleteCluster(cmd *cobra.Command, args []string) error {
-	if cmdClusterID == "" {
-		return errors.New(color.RedString("Please select a cluster to delete"))
-	}
+// validationOutput runs our pre-checks.
```
Comment (rossf7) [inline commands/delete_cluster.go]: Nitpick: it's not exported but I'd like it if this was `deleteClusterValidationOutput` to match GoDoc convention

File: commands/delete_cluster.go [hunk 2]
```diff
@@ -43,58 +69,104 @@ func init() {
...
+	if err != nil {
+		var headline = ""
+		var subtext = ""
+
+		switch err.Error() {
+		case "":
+			return
+		case errNotLoggedIn:
+			headline = "You are not logged in."
+			subtext = fmt.Sprintf("Use '%s login' to login or '--auth-token' to pass a valid auth token.", config.ProgramName)
```
Comment (rossf7) [inline commands/delete_cluster.go]: I <3 these prompts for what the user should do

File: commands/delete_cluster.go [hunk 3]
```diff
@@ -43,58 +69,104 @@ func init() {
...
+		// print output
+		fmt.Println(color.RedString(headline))
+		if subtext != "" {
+			fmt.Println(subtext)
+		}
+		os.Exit(1)
 	}
+}
 
+// validatePreConditions checks preconditions and returns an error in case
```
Comment (rossf7) [inline commands/delete_cluster.go]: Another GoDoc nitpick

File: commands/delete_cluster_test.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// TestDeleteClusterSuccess runs test case that are supposed to succeed
+func TestDeleteClusterSuccess(t *testing.T) {
+	// mock server always responding positively
```
Comment (rossf7) [inline commands/delete_cluster_test.go]: Really nitpicky but "responds" reads better to me

File: commands/delete_cluster_test.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		t.Log("mockServer request: ", r.Method, r.URL)
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`{"code": "RESOURCE_DELETION_STARTED", "message": "We'll soon nuke this cluster"}`))
```
Comment (rossf7) [inline commands/delete_cluster_test.go]: 👍  for nuking clusters from orbit

**Comment by marians** [comment]
Fair enough! Comments and docstrings are as important as the executed code!

**Pull request merged**